tpm2/test/tpm2_test.go

```diff
@@ -419,6 +421,79 @@ func TestHash(t *testing.T) {
 	})
 }
 
+func testHashSequence(t *testing.T, rw io.ReadWriter, hierarchy tpmutil.Handle, hashAlg Algorithm, data []byte) ([]byte, *Ticket) {
+	const (
+		maxDigestBuffer = 1024
+		seqAuth         = ""
+	)
+
+	seq, err := HashSequenceStart(rw, seqAuth, hashAlg)
+	if err != nil {
+		t.Fatalf("HashSequenceStart failed: %v", err)
+	}
+
+	defer tpm2.FlushContext(rw, seq)
+
+	for len(data) > maxDigestBuffer {
+		err = SequenceUpdate(rw, seqAuth, seq, data[:maxDigestBuffer])
+		if err != nil {
+			t.Fatalf("SequenceUpdate failed: %v", err)
+		}
+
+		data = data[maxDigestBuffer:]
+	}
+
+	digest, ticket, err := SequenceComplete(rw, seqAuth, seq, hierarchy, data)
+	if err != nil {
+		t.Fatalf("SequenceComplete failed: %v", err)
+	}
+
+	return digest, ticket
+}
+
+func TestHashSequence(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	run := func(t *testing.T, data []byte, hierarchy tpmutil.Handle, wantValidation bool) {
+		gotDigest, gotValidation := testHashSequence(t, rw, hierarchy, AlgSHA256, data)
+		wantDigest := sha256.Sum256(data)
+
+		if !bytes.Equal(gotDigest, wantDigest[:]) {
+			t.Errorf("Hash(%q) returned digest %x, want %x", data, gotDigest, wantDigest)
+		}
+		if wantValidation && len(gotValidation.Digest) == 0 {
```

Comment:
This looks correct.

If I recall correctly, `gotValidation.Digest` isn't the same as the hash digest, it's an HMAC value. So it makes sense that we don't compare the value to `gotDigest`.